pkg/controller/pxc/status.go

```diff
@@ -319,32 +205,36 @@ func (r *ReconcilePerconaXtraDBCluster) appStatus(app api.StatefulApp, podSpec *
 	}
 
 	for _, pod := range list.Items {
+		for _, cntr := range pod.Status.ContainerStatuses {
+			if cntr.State.Waiting != nil && cntr.State.Waiting.Message != "" {
+				status.Message += cntr.Name + ": " + cntr.State.Waiting.Message + "; "
+			}
+		}
+
 		for _, cond := range pod.Status.Conditions {
 			switch cond.Type {
 			case corev1.ContainersReady:
-				if cond.Status == corev1.ConditionTrue {
-					if !isPXC(app) {
-						status.Ready++
-					} else {
-						isPodReady := true
-						if cr170OrGreater {
-							isPodWaitingForRecovery, _, err := r.isPodWaitingForRecovery(namespace, pod.Name)
-							if err != nil {
-								return api.AppStatus{}, fmt.Errorf("parse %s pod logs: %v", pod.Name, err)
-							}
-							isPodReady = !isPodWaitingForRecovery && pod.ObjectMeta.Labels["controller-revision-hash"] == sfs.Status.UpdateRevision
-						}
-
-						if isPodReady {
-							status.Ready++
-						}
-					}
-				} else if cond.Status == corev1.ConditionFalse {
-					for _, cntr := range pod.Status.ContainerStatuses {
-						if cntr.State.Waiting != nil && cntr.State.Waiting.Message != "" {
-							status.Message += cntr.Name + ": " + cntr.State.Waiting.Message + "; "
-						}
-					}
+				if cond.Status != corev1.ConditionTrue {
+					continue
+				}
+
+				if !isPXC(app) {
+					status.Ready++
+					continue
+				}
+
+				if !cr170OrGreater {
+					status.Ready++
+					continue
+				}
```

Comment:
maybe 
```go
				if !isPXC(app) || lt17 {
					status.Ready++
					continue
				}
```
instead? 

_*taking into account the previous comment about logic inversion for that bool._
_** and don't forget to change login in the caller is so. I mean `cr.CompareVersionWith("1.7.0") >= 0` to `cr.CompareVersionWith("1.7.0") == -1` in L168_

Reply:
see https://github.com/percona/percona-xtradb-cluster-operator/commit/337d9d567bbad589cfd8d851d66960f2a55d8ac0

I used `crLt170` as variable name for clarity, though